main.go

```diff
@@ -150,6 +150,19 @@ func mustGetProfilePath() string {
 	return filepath.Join(mustGetConfigPath(), globalMinioProfilePath)
 }
 
+func setupProfilingFromEnv(profiler *interface {
+	Stop()
+}) {
+	switch os.Getenv("MINIO_PROFILER") {
```

Comment:
The reason the switch-case was moved into app,BeforeFunc is to delay starting of profiling until configDir is initialised. That happens inside app.BeforeFunc. If we moved the switch-case outside then we need to move setting of configDir from command line flags outside too.